bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java

```diff
@@ -277,9 +294,54 @@ public String getRoleAttribute() {
 
     @Override
     public String getAction() {
+        List<String> supportedSubmitActions = this.getSupportedSubmitActions();
+        String resourceType = resource.getValueMap().get("sling:resourceType", String.class);
+        supportedSubmitActions = Arrays.asList("spreadsheet");
+        if (supportedSubmitActions.contains(resource.getValueMap().get(ReservedProperties.PN_SUBMIT_ACTION_NAME))) {
+            if (resourceType != null && resourceType.contains("/franklin")) {
+                return "";
+            } else {
+                return "https://forms.adobe.com" + ADOBE_GLOBAL_API_ROOT + FORMS_RUNTIME_API_GLOBAL_ROOT + "/submit/" +
+                    ComponentUtils.getEncodedPath(resource.getPath() + ".model.json");
+            }
+        }
         return getContextPath() + ADOBE_GLOBAL_API_ROOT + FORMS_RUNTIME_API_GLOBAL_ROOT + "/submit/" + getId();
     }
 
+    private List<String> getSupportedSubmitActions() {
+        String supportedSubmitActionsUrl = "https://forms.adobe.com/adobe/forms/af/supportedSubmitActions";
```

Comment:
Can we move this to some other function in ComponentUtils ? I would want the sling model code to be short

Reply:
Ok @rismehta , will move this

Reply:
moved